src/emailjs-tcp-socket.js

```diff
@@ -619,8 +662,16 @@
                 // process the data available on the socket
                 self._onData(readInfo.data);
 
-                // queue the next read
-                self._readLegacySocket();
+                // Queue the next read.
+                // If a STARTTLS handshake might be upcoming, postpone this onto
+                // the task queue so the IMAP client has a chance to call upgradeToSecure;
+                // without this, we might eat the beginning of the handshake.
+                // If we are already secure, just call it (for performance).
+                if (self.ssl) {     // are we secure yet?
+                    self._readLegacySocket();
+                } else {
+                    setZeroTimeout(self._readLegacySocket.bind(self));
```

Comment:
for future reference: 
this works because for [an expected STARTTLS we withhold any actual activity on the socket](https://github.com/emailjs/emailjs-imap-client/blob/6da06bb96558422e4b365e3d39f127c54c653c45/src/emailjs-imap-client.js#L893-L897)

this works probably because the stars are aligned in our favor. i am *not* sure this always works. [i asked those who will know better than me](https://twitter.com/felixhammerl/status/706983424406495234) if that's a sure thing or not. if `foo` in promise.resolve(foo) is also pushed to the next run loop iteration, i'd assume things may get messy ... if we do it this way, we should add a big fat pointer to the imap and smtp clients :)